content/en/docs/tasks/administer-cluster/change-pv-access-mode-readwriteoncepod.md

```diff
@@ -0,0 +1,184 @@
+
+{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
+
+{{< note >}}
+The ReadWriteOncePod access mode is only supported for
+{{< glossary_tooltip text="CSI" term_id="csi" >}} volumes.
+To use this volume access mode you will need to update the following
+[CSI sidecars](https://kubernetes-csi.github.io/docs/sidecar-containers.html)
+to these versions or greater:
+
+* [csi-provisioner:v3.0.0+](https://github.com/kubernetes-csi/external-provisioner/releases/tag/v3.0.0)
+* [csi-attacher:v3.3.0+](https://github.com/kubernetes-csi/external-attacher/releases/tag/v3.3.0)
+* [csi-resizer:v1.3.0+](https://github.com/kubernetes-csi/external-resizer/releases/tag/v1.3.0)
+{{< /note >}}
+
+## Why should I use ReadWriteOncePod?
+
+Prior to Kubernetes 1.22, the ReadWriteOnce access mode was commonly used to
+restrict PersistentVolume access for workloads that required single-writer
+access to storage. However, this access mode had a limitation: it restricted
+volume access to a single *node*, allowing multiple pods on the same node to
+read from and write to the same volume simultaneously. This could pose a risk
+for applications that demand strict single-writer access for data safety.
+
+If ensuring single-writer access is critical for your workloads, consider
+migrating your volumes to ReadWriteOncePod.
+
+<!-- steps -->
+
+## Migrating existing PersistentVolumes
+
+If you have existing PersistentVolumes, they can be migrated to use
+ReadWriteOncePod.
+
+In this example, there is already a "cat-pictures-pvc" PersistentVolumeClaim
+that is bound to a "cat-pictures-pv" PersistentVolume, and a
+"cat-pictures-writer" Deployment that uses this PersistentVolumeClaim.
+
+```yaml
+---
+# You may omit this object if your storage plugin supports dynamic provisioning.
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: cat-pictures-pv
+spec:
+  storageClassName: standard
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 1Gi
+  csi:
+    driver: io.kubernetes.storage.mock
+    volumeHandle: cat-pictures-pv
+
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: cat-pictures-pvc
+spec:
+  storageClassName: standard
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+  volumeName: cat-pictures-pv # Omit this field if using dynamic provisioning.
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cat-pictures-writer
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: cat-pictures-writer
+  template:
+    metadata:
+      labels:
+        app: cat-pictures-writer
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+      volumes:
+      - name: cat-pictures
+        persistentVolumeClaim:
+          claimName: cat-pictures-pvc
+          readOnly: false
+```
+
+{{< note >}}
+If your storage plugin supports
+[Dynamic provisioning](/docs/concepts/storage/dynamic-provisioning/),
```

Comment:
@xing-yang Should leave the PersistentVolume resource in the manifest above? I was thinking yes to support drivers without dynamic provisioning. But this requires additional caveats (e.g. the driver details in the PV will differ). What are your thoughts?

Reply:
After some review I decided to remove the PersistentVolume resource and make the example use dynamic provisioning. I think this simplifies the instructions and makes it easier to read and understand.

Reply:
Sounds good.